pkg/cloud/services/ec2/console_test.go

```diff
@@ -0,0 +1,142 @@
+		check      func(data string, err error)
+	}{
+		{
+			name:       "Should return decoded instance output if instanceID exist",
+			instanceID: "1234",
+			expect: func(m *mock_ec2iface.MockEC2APIMockRecorder) {
+				m.GetConsoleOutput(gomock.Eq(&ec2.GetConsoleOutputInput{
+					InstanceId: aws.String("1234"),
+					Latest:     aws.Bool(true),
+				})).Return(&ec2.GetConsoleOutputOutput{
+					Output: aws.String("iVBORw0KGgoAAAANSUhEUgAAAGQAAABkCAYA"),
+				}, nil)
+			},
+			check: func(data string, err error) {
+				if err != nil {
+					t.Fatalf("did not expect err: %v", err)
+				}
+				expected, _ := base64.StdEncoding.DecodeString(aws.StringValue(aws.String("iVBORw0KGgoAAAANSUhEUgAAAGQAAABkCAYA")))
+				if data != string(expected) {
+					t.Fatalf("expected: %+v, but got : %+v", string(expected), data)
+				}
+			},
+		},
+		{
+			name:       "Should return with error if AWS fails with dependency issue",
+			instanceID: "1234",
+			expect: func(m *mock_ec2iface.MockEC2APIMockRecorder) {
+				m.GetConsoleOutput(gomock.Eq(&ec2.GetConsoleOutputInput{
+					InstanceId: aws.String("1234"),
+					Latest:     aws.Bool(true),
+				})).Return(nil, awserrors.NewFailedDependency("dependency failure"))
+			},
+			check: func(data string, err error) {
+				if err == nil {
+					t.Fatalf("expected an error but got nothing")
+				}
+				if data != "" {
+					t.Fatalf("did not expect data, but got %+v", data)
+				}
+			},
+		},
+		{
+			name:       "Should return with error if instance output decoding fails",
+			instanceID: "1234",
+			expect: func(m *mock_ec2iface.MockEC2APIMockRecorder) {
+				m.GetConsoleOutput(gomock.Eq(&ec2.GetConsoleOutputInput{
+					InstanceId: aws.String("1234"),
+					Latest:     aws.Bool(true),
+				})).Return(&ec2.GetConsoleOutputOutput{
+					Output: aws.String("abc"),
+				}, nil)
+			},
+			check: func(data string, err error) {
+				if err == nil {
+					t.Fatalf("expected an error")
+				}
+				if data != "" {
+					t.Fatalf("did not expect data, but got %+v", data)
+				}
+			},
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			scheme, err := setupScheme()
+			if err != nil {
+				t.Fatalf("failed during scheme registration with err: %+v", err)
+			}
+
+			client := fake.NewClientBuilder().WithScheme(scheme).Build()
+			mockEC2Client := mock_ec2iface.NewMockEC2API(mockCtrl)
+			mockSSMClient := mock_ssmiface.NewMockSSMAPI(mockCtrl)
+
+			csScope, err := scope.NewClusterScope(scope.ClusterScopeParams{
+				Cluster: &clusterv1.Cluster{
+					ObjectMeta: metav1.ObjectMeta{
+						Namespace: "ns",
+						Name:      "clusterName",
+					},
+				},
+				Client:     client,
+				AWSCluster: newAWSCluster(),
+			})
+			if err != nil {
+				t.Fatalf("Failed during scope init")
+			}
+
+			s := &Service{
+				scope:     csScope,
+				EC2Client: mockEC2Client,
+				SSMClient: mockSSMClient,
+			}
+
+			tc.expect(mockEC2Client.EXPECT())
+			tc.check(s.GetConsoleOutput(tc.instanceID))
```

Comment:
Are we using this method anywhere?

Reply:
Thanks for catching this, idk how I missed it :( 
So shall I remove this file all together?

Reply:
Even I have seen it now, if it is not used, it's good to clean this up now

Reply:
This is an old code, we could clean up.

Reply:
Ya sure I will do it